Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -394,6 +403,7 @@ void UPSWorldSubsystem::SaveDataAsync()
 		return;
 	}
 
+	OnCurrentScoreChanged.Broadcast();
```

Comment:
Your new delegate behaves like Pre-event-happened, instead of Post-event-happened.

I believe, calling it before save is loaded\created is too early. For such early events, or when save is not created, other systems should use `OnInitialize` delegate, while `OnCurrentScoreChanged.Broadcast()` should broadcast at the end of the save has been loaded\created.

In this way, other systems will have less problems with progression data they use

Reply:
SaveDataAsync is the latest step before the data will be actually saved to the disk, so this post-event